apps/ex_nvr/lib/ex_nvr/model/remote_storage.ex

```diff
@@ -0,0 +1,123 @@
+defmodule ExNVR.Model.RemoteStorage do
+  use Ecto.Schema
+
+  alias Ecto.Changeset
+
+  @type t :: %__MODULE__{
+          type: :s3 | :http,
+          config: Config.t() | nil,
+          inserted_at: DateTime.t() | nil,
+          updated_at: DateTime.t() | nil
+        }
+
+  defmodule Config do
+    use Ecto.Schema
+
+    alias Ecto.Changeset
+
+    @type t :: %__MODULE__{
+            url: binary(),
+            username: binary(),
+            password: binary(),
+            token: binary(),
+            bucket: binary(),
+            access_key_id: binary(),
+            secret_access_key: binary(),
+            region: binary()
+          }
+
+    @primary_key false
+    embedded_schema do
+      field :url, :string
```

Comment:
`url` should be in the top schema, not embedded. It's optional for `s3` and mandatory for plain `http`.